### PR TITLE
[NEW] Settings to enable E2E encryption for Private and Direct Rooms by default

### PR DESCRIPTION
**Pull request by rodrigok**

### CODE REVIEW

File: app/lib/server/functions/createDirectRoom.js [hunk 1]
```diff
@@ -1,45 +1,81 @@
 import { Rooms, Subscriptions } from '../../../models/server';
+import { getDefaultSubscriptionPref } from '../../../utils';
+import { callbacks } from '../../../callbacks';
```
Comment (sampaiodiego) [inline app/lib/server/functions/createDirectRoom.js]: this way editor's ctrl+click works =)
```suggestion
import { getDefaultSubscriptionPref } from '../../../utils/server';
import { callbacks } from '../../../callbacks/server';
```

File: app/e2e/server/beforeCreateRoom.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { callbacks } from '../../callbacks';
+import { settings } from '../../settings';
```
Comment (sampaiodiego) [inline app/e2e/server/beforeCreateRoom.js]:
```suggestion
import { callbacks } from '../../callbacks/server';
import { settings } from '../../settings/server';
```

File: app/cas/server/cas_server.js [hunk 3]
```diff
@@ -11,8 +10,9 @@ import CAS from 'cas';
 
 import { logger } from './cas_rocketchat';
 import { settings } from '../../settings';
-import { Rooms, Subscriptions, CredentialTokens } from '../../models';
+import { Rooms, CredentialTokens } from '../../models';
```
Comment (sampaiodiego) [inline app/cas/server/cas_server.js]:
```suggestion
import { Rooms, CredentialTokens } from '../../models/server';
```

**Comment by reetp** [comment]
Why this was merged when:

a) It is still considered Beta
""E2E_Enable_alert": "This feature is currently in beta!"

b) It is still not supported on any of the mobile apps?
https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/141

I just don't get the logic of enforcing this by default when the code is not considered stable, and the apps have no support.

Also noted that these probably still apply (for those who do not read documentation)

- Bots may not be able to see encrypted messages until they implement support for it.
- Uploads will not be encrypted in this version.

**Comment by rodrigok** [comment]
@reetp this is not enabled by default, but a setting to be able to enable it by default.

There are some clients that want this feature since it doesn't hurt anyone we added the option.

Yes, it's still in beta, it's not supported by our mobile apps yet, but if you need this and use only the web clients you may use this new setting.

**Comment by reetp** [comment]
> @reetp this is not enabled by default, but a setting to be able to enable it by default.
> 
KK - sorry but the heading was not very clear.

> There are some clients that want this feature since it doesn't hurt anyone we added the option.
> 

Indeed

> Yes, it's still in beta, it's not supported by our mobile apps yet, but if you need this and use only the web clients you may use this new setting.

Yup.

Thanks for the clarification.

**Pull request merged**